src/common/common.test.ts

```diff
@@ -260,91 +465,837 @@ describe("concatenateContexts", () => {
+      );
+    });
+  });
+
+  describe("throws if the dereferenced data is invalid JSON", () => {
+    let mockedFetch: jest.MockedFunction<(typeof UniversalFetch)["fetch"]>;
+
+    beforeEach(() => {
+      mockedFetch = jest
+        .fn<(typeof UniversalFetch)["fetch"]>()
+        .mockResolvedValueOnce(new Response("Not JSON")) as jest.MockedFunction<
+        (typeof UniversalFetch)["fetch"]
+      >;
+    });
+
+    it("returnLegacyJsonld: true", async () => {
+      await expect(
+        getVerifiableCredential(
+          "https://example.org/ns/someCredentialInstance",
+          {
+            fetch: mockedFetch as (typeof UniversalFetch)["fetch"],
+          },
+        ),
+      ).rejects.toThrow(
+        "Parsing the Verifiable Credential [https://example.org/ns/someCredentialInstance] as JSON failed: SyntaxError:",
+      );
+    });
+    it("returnLegacyJsonld: false", async () => {
+      await expect(
+        getVerifiableCredential(
+          "https://example.org/ns/someCredentialInstance",
+          {
+            fetch: mockedFetch as (typeof UniversalFetch)["fetch"],
+            returnLegacyJsonld: false,
+          },
+        ),
+      ).rejects.toThrow(
+        "Parsing the Verifiable Credential [https://example.org/ns/someCredentialInstance] as JSON failed: SyntaxError:",
+      );
     });
-    expect(mockedFetch).toHaveBeenCalledWith("https://some.vc");
   });
 
-  it("throws if the VC ID cannot be dereferenced", async () => {
+  describe("throws if the dereferenced data is not a VC", () => {
+    let mockedFetch: jest.MockedFunction<(typeof UniversalFetch)["fetch"]>;
+
+    beforeEach(() => {
+      mockedFetch = jest
+        .fn<(typeof UniversalFetch)["fetch"]>()
+        .mockResolvedValueOnce(
+          new Response(JSON.stringify({ something: "but not a VC" })),
+        ) as jest.MockedFunction<(typeof UniversalFetch)["fetch"]>;
+    });
+
+    it("returnLegacyJsonld: true", async () => {
+      await expect(
+        getVerifiableCredential(
+          "https://example.org/ns/someCredentialInstance",
+          {
+            fetch: mockedFetch as (typeof UniversalFetch)["fetch"],
+          },
+        ),
+      ).rejects.toThrow(
+        "The value received from [https://example.org/ns/someCredentialInstance] is not a Verifiable Credential",
+      );
+    });
+    it("returnLegacyJsonld: false", async () => {
+      await expect(
+        getVerifiableCredential(
+          "https://example.org/ns/someCredentialInstance",
+          {
+            fetch: mockedFetch as (typeof UniversalFetch)["fetch"],
+            returnLegacyJsonld: false,
+          },
+        ),
+      ).rejects.toThrow(
+        "Verifiable credential is not an object, or does not have an id",
+      );
+    });
+  });
+
+  it.skip("throws if the dereferenced data has an unsupported content type", async () => {
```

Comment:
Why is this disabled?

Reply:
Because this would be a new check (the code for this is also currently deleted and/or disabled) - I suggest we enabled it in the next major version.